pkg/puller/puller.go

```diff
@@ -134,7 +133,7 @@ func (p *Puller) manage(ctx context.Context, warmupTime time.Duration) {
 		prevRadius = syncRadius
 
 		_ = p.topology.EachPeerRev(func(addr swarm.Address, po uint8) (stop, jumpToNext bool, err error) {
-			if po >= neighborhoodDepth {
+			if po >= syncRadius {
```

Comment:
So there are 2 cases in the issue mentioned. Does this handle both the cases?

Reply:
the other case i wanted to tackle in a separate PR, I can do it in this one, but the PR will be bigger.